packages/hdwallet-core/src/ethereum.ts

```diff
@@ -107,6 +107,12 @@ export interface ETHWalletInfo extends HDWalletInfo {
    */
   ethSupportsNetwork(chain_id: number): Promise<boolean>;
 
+  /**
+   * Get the current chainId from ethereum's JSON RPC
+   * https://eips.ethereum.org/EIPS/eip-695
+   */
+  ethGetChainId?(): Promise<number | void>;
```

Comment:
Or null, or whatever, but not void -- you can return anything, including a number, from a function whose type is assignable to a function whose return type is void. More to the point, though, if the method's optional already and we don't need to determine at runtime if the question is valid (which I don't think we do?) the return type shouldn't be ambiguous.

```suggestion
  ethGetChainId?(): Promise<number | undefined>;
```